src/main/java/uk/gov/gchq/koryphe/impl/binaryoperator/Product.java

```diff
@@ -39,4 +45,19 @@ protected Long aggregateLong(final Long a, final Long b) {
     protected Double aggregateDouble(final Double a, final Double b) {
         return a * b;
     }
+
+    @Override
+    protected Float aggregateFloat(final Float a, final Float b) {
+        return a * b;
```

Comment:
As an aside, is it worth warning about precision errors here? Would we want to support aggregation of BigDecimal and BigInteger?